edb/server/connpool/pool.py

```diff
@@ -224,6 +236,37 @@ def _wakeup_next_waiter(self) -> None:
                 waiter.set_result(None)
                 break
 
+    def log_connection(self, event: str, timestamp: float = 0) -> None:
+        if not timestamp:
+            timestamp = time.monotonic()
+        if self._is_log_batching:
+            self._log_events[event] = self._log_events.setdefault(event, 0) + 1
+        elif timestamp - self._last_log_timestamp > MIN_LOG_TIME_THRESHOLD:
```

Comment:
There can be a race here, as asyncio scheduled calls can be executed slightly past their time

```suggestion
        elif (
            timestamp - self._last_log_timestamp > MIN_LOG_TIME_THRESHOLD
            and not self._is_log_batching
        ):
```

Reply:
Right, thanks for the tip! The `elif` should've taken care of that. Let me add a comment

Reply:
Of course, you're right! 🤦‍♂️